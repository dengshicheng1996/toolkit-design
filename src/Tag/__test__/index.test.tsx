import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Tag } from '..'

describe('Tag', () => {
  it('snapshot', () => {
    const component = shallow(
      <Tag className="a b" data-y="111">
        button
      </Tag>,
    )
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('snapshot 2', () => {
    const component = shallow(
      <Tag className="a b" data-count="3" type="line" theme="orange">
        button
      </Tag>,
    )
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('onPress', () => {
    const onPress = jest.fn()
    const component = shallow(
      <Tag className="a b" type="line" theme="orange" onPress={onPress}>
        button
      </Tag>,
    )

    component.simulate('press')

    expect(onPress).toHaveBeenCalled()
  })
})
