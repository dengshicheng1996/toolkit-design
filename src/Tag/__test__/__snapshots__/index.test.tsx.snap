// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag snapshot 1`] = `
<Touchable>
  <span
    className="amd-tag amd-tag-filling-normal a b"
    data-y="111"
  >
    button
  </span>
</Touchable>
`;

exports[`Tag snapshot 2 1`] = `
<Touchable>
  <span
    className="amd-tag amd-tag-line-orange a b"
    data-count="3"
  >
    button
  </span>
</Touchable>
`;
